### ISSUE TITLE
Build failure with GHC 8.0 and 8.2

### ISSUE THREAD

**Issue by snoyberg** [iand675/metrics#11]
I'm seeing this build failure:

```
[ 7 of 20] Compiling Data.Metrics.Reservoir.Uniform ( src/Data/Metrics/Reservoir/Uniform.hs, dist/build/Data/Metrics/Reservoir/Uniform.o )

src/Data/Metrics/Reservoir/Uniform.hs:46:1: error:
    • Illegal instance declaration for
        ‘HasInnerReservoir UniformReservoir a’
        The coverage condition fails in class ‘HasInnerReservoir’
          for functional dependency: ‘s -> a’
        Reason: lhs type ‘UniformReservoir’ does not determine rhs type ‘a’
        Un-determined variable: a_aukW
        Using UndecidableInstances might help
    • In the instance declaration for
        ‘HasInnerReservoir UniformReservoir a_aukW’
   |
46 | makeFields ''UniformReservoir
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

**Comment by iand675** [comment]
@snoyberg is there anywhere I can see the failure publicly? I've built metrics locally on two separate machines using GHC 8.0 and 8.2, so I'm a little confused about what your configuration is doing differently.

**Comment by snoyberg** [comment]
I was able to reproduce with:

    stack build --resolver nightly-2017-08-08 metrics-0.4.1.0

**Comment by iand675** [comment]
OK, thanks. Should be legitimately fixed in `metrics-0.4.1.1`
